antismash/modules/rrefinder/__init__.py

```diff
@@ -0,0 +1,159 @@
+# License: GNU Affero General Public License v3 or later
+# A copy of GNU AGPL v3 should have been included in this software package in LICENSE.txt.
+
+""" Implementation of RREFinder in antiSMASH.
+    RiPP Recognition Elements (RREs) are protein domains of ~100
+    amino acids in RiPP modifying enzymes that are vital for RiPP
+    precursor peptide recognition. They are found in roughly half
+    of all bacterial RiPP classes.
+    RREFinder is a tool for the recognition of these domains.
+    RREFinder has two modes for RRE detection: precision
+    mode, which detects RREs of known RiPP classes with high
+    confidence using a library of pHMMs, and exploratory mode,
+    which uses HHPred to detect RREs. Exploratory mode detects
+    more false positives, but is capable of detecting RREs for
+    which no specific pHMM model has been designed.
+
+    Only precision mode has been implemented here.
+
+    For more details, see:
+    https://msystems.asm.org/content/5/5/e00267-20/article-info
+    and
+    https://github.com/Alexamk/RREFinder
+"""
+
+from typing import Any, Dict, List, Optional
+
+from antismash.common.secmet import Record
+from antismash.config import ConfigType
+from antismash.config.args import ModuleArgs
+
+from antismash.common.hmmer import ensure_database_pressed
+from antismash.common import path
+
+from .html_output import generate_html, will_handle
+from .rrefinder import run_rrefinder, RREFinderResults
+
+NAME = "rrefinder"
+SHORT_DESCRIPTION = "Module for the pHMM-based detection of RiPP Recognition Elements (RREs)"
+
+
+def get_arguments() -> ModuleArgs:
+    """ Builds any commandline argument constructs that may be required
+
+        Returns:
+            an empty or populated ModuleArgs instance
+    """
+
+    args = ModuleArgs('RREfinder options', 'rre')
+    args.add_analysis_toggle('run',
```

Comment:
It would be more consistent with other analysis modules like active site finder if the analysis toggle was `--rre` and not `--rre-run`.

Reply:
I guess this is because unlike active site finder, there's also options that start on `--rre`?

Reply:
That can be changed, it would make it consistent with `--fullhmmer` and `--fullhmmer-pfamdb-version`.